packages/@aws-cdk/aws-fsx/README.md

```diff
@@ -17,8 +17,171 @@
+you to spin up and run a battle-tested high-performance file system in minutes. It also provides multiple deployment
+options so you can optimize cost for your needs.
+
+Amazon FSx for Lustre is POSIX-compliant, so you can use your current Linux-based applications without having to make
+any changes. Amazon FSx for Lustre provides a native file system interface and works as any file system does with your
+Linux operating system. It also provides read-after-write consistency and supports file locking.
+
+### Installation
+
+Import to your project:
 
 ```ts
 import fsx = require('@aws-cdk/aws-fsx');
 ```
+
+### Basic Usage
+
+Setup required properties and create:
+
+```ts
+const stack = new Stack(app, 'Stack');
+const vpc = new Vpc(stack, 'VPC');
+
+const fileSystem = new LustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration: { deploymentType: LustreDeploymentType.SCRATCH_2 },
+  storageCapacityGiB: 1200,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
+```
+
+### Connecting
+
+To control who can access the file system, use the `.connections` attribute. FSx has a fixed default port, so you don't
+need to specify the port. This example allows an EC2 instance to connect to a file system:
+
+```ts
+fileSystem.connections.allowDefaultPortFrom(instance);
+```
+
+### Mounting
+
+To mount an FSx for Lustre file system programmatically, the mount name of the file system needs to be determined. The
+easiest way to do that is with the AWS CLI, which can be written into a User Data script to be run on start-up of your
+instance. The following example can be used for reference on how to bring up a file system and EC2 instance with the
+file system mounted on the instance at start-up:
+
+```ts
+const app = new App();
+const stack = new Stack(app, 'AwsCdkFsxLustre');
+const vpc = new Vpc(stack, 'VPC');
+
+const lustreConfiguration = {
+  deploymentType: LustreDeploymentType.SCRATCH_2,
+};
+const fs = new LustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration,
+  storageCapacityGiB: 1200,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
+
+const inst = new Instance(stack, 'inst', {
+  instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.LARGE),
+  machineImage: new AmazonLinuxImage({
+    generation: AmazonLinuxGeneration.AMAZON_LINUX_2
+  }),
+  vpc,
+  vpcSubnets: {
+    subnetType: SubnetType.PUBLIC,
+  }
+});
+inst.connections.securityGroups.forEach((securityGroup => {
+    fs.connections.addSecurityGroup(securityGroup);
+}));
+// Need to give the instance access to read information about FSx to determine the file system's mount name.
+inst.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName("AmazonFSxReadOnlyAccess"));
+
+// Add a dependency on the file system to the instance so that we can be sure the file system is instantiated before querying
+// for its mount name.
+if (fs.node.defaultChild) {
```

Comment:
Is this necessary though? CloudFormation should be taking care of this automatically.

Reply:
It doesn't automatically create the dependency, but it does seem to create the EC2 instance after the file system is created. I'll remove this.

Reply:
Use of a `{ Ref }` or `{ Fn::GetAtt }` will automatically create a dependency. You only need to use `DependsOn` if the dependency is implicit, because creating the resource has some side effect (such as `Policy`s usually do).